crates/biome_js_analyze/src/lint/nursery/no_document_import_in_page.rs

```diff
@@ -0,0 +1,94 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::{JsFileSource, JsImport};
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Prevents importing `next/document` outside of `pages/_document.jsx` in Next.js projects.
+    ///
+    /// The `next/document` module is intended for customizing the document structure globally in Next.js.
+    /// Importing it outside of `pages/_document.js` can cause unexpected behavior and break certain features of the framework.
+    ///
+    /// ## Examples
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// import { Document, Html } from 'next/document'
+    ///
+    /// export default class MyDocument extends Document {
+    ///   render() {
+    ///     return (
+    ///       <Html lang="en">
+    ///         {/* */}
+    ///       </Html>
+    ///     )
+    ///   }
+    /// }
+    ///
+    pub NoDocumentImportInPage {
+        version: "next",
+        name: "noDocumentImportInPage",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("no-document-import-in-page")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
+}
+
+impl Rule for NoDocumentImportInPage {
+    type Query = Ast<JsImport>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        if !ctx.source_type::<JsFileSource>().is_jsx() {
+            return None;
+        }
+
+        let import = ctx.query();
+        let import_source = import.import_clause().ok()?.source().ok()?;
+        let module_name = import_source.inner_string_text().ok()?;
+
+        if module_name != "next/document" {
+            return None;
+        }
+
+        let path = ctx.file_path();
+
+        if !path
+            .ancestors()
+            .filter_map(|a| a.file_name())
+            .any(|f| f == "pages")
+        {
+            return None;
+        }
+
+        let file_name = path.file_stem()?.to_str()?;
+        let parent_name = path.parent()?.file_stem()?.to_str()?;
+
+        if parent_name == "_document" || file_name == "_document" {
+            return None;
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        return Some(
+            RuleDiagnostic::new(
```

Comment:
You might want to explain the users why this error occurs ( See [Rule pllars](https://biomejs.dev/linter/#rule-pillars) in this case.
In [eslint-plugin-next/no-document-import-in-page](https://github.com/vercel/next.js/blob/canary/packages/eslint-plugin-next/src/rules/no-document-import-in-page.ts), they are not explaining why the error is triggered but providing the link [Why This Error Occurred](https://nextjs.org/docs/messages/no-head-import-in-document#why-this-error-occurred).
We could add an explanation IMHO.

Reply:
The explanation in their link is pretty vague 😅

"to avoid unexpected behaviors" was the best I could extract from it.

Reply:
Yeah, Next.js documentation is usually bad 😔

Reply:
I agree